installation/alpine.sh

```diff
@@ -0,0 +1,591 @@
+    then
+        return 1
+    fi
+
+    echo
+
+    # Print top box row
+    for i in $(seq ${width})
+    do
+        echo -n "#"
+    done
+    echo
+
+    # Print message with borders
+    echo -n "# "; echo -n "$1" ; echo " #"
+
+    # Print bottom box row
+    for i in $(seq ${width})
+    do
+        echo -n "#"
+    done
+    echo
+}
+
+# Check OS and version are supported.
+#
+# params: nothing
+#
+# return:
+# - exit 1 if OS isn't supported
+# - exit 2 OS version isn't supported
+check_os()
+{
+	os_ver_file=/etc/alpine-release
+
+	if [ ! -e "$os_ver_file" ]
+	then
+		echo "OS not supported! Please run it on Alpine Linux."
+		exit 1
+	fi
+
+	if ! grep -qE "^$alpine_ver" "$os_ver_file"
+	then
+		echo "Alpine $(cat $os_ver_file) not supported: need Alpine $alpine_ver."
+		exit 2
+	fi
+}
+
+# Add Alpine community repo and update packages list.
+#
+# params: nothing
+#
+# return: nothing
+update_repo()
+{
+    # Enable community repository if not already enabled
+    if ! grep -qE '^http.*\/community' /etc/apk/repositories
+    then
+        echo "Adding Alpine community repository..."
+        echo "http://dl-cdn.alpinelinux.org/alpine/v${alpine_ver}/community" >> /etc/apk/repositories
+    fi
+
+    # Update packages list and install NodeJS
+    echo "Updating packages list..."
+    apk update
+}
+
+# Fetch source code needed for the project's building.
+# If the script runs in dev mode, the source code must be
+# in $tmp_dir diretory. Else, the code is downloaded from GitHub.
+#
+# params: nothing
+#
+# return: 
+# - exit 1 if dev mode but no source code in tmp path
+get_source_code()
+{
+    # If not in dev mode
+    if [ "$dev" -eq 0 ]
+    then
+        # Download and extract repo release
+        echo "Downloading and extracting Immich v$immich_ver..."
+        wget https://github.com/immich-app/immich/archive/refs/tags/v${immich_ver}.tar.gz -O ${tmp_dir}/immich-${immich_ver}.tar.gz || exit
+        tar -xzf ${tmp_dir}/immich-${immich_ver}.tar.gz --directory ${tmp_dir}
+    else
+        # Dev source code must be in temp directory
+        echo "Getting source code from local directory..."
+        if [ ! -d "$tmp_dir/immich-$immich_ver" ]
+        then
+            echo "You need to put $immich_ver development code in $tmp_dir/immich-$immich_ver/".
+            exit 1
+        fi
+    fi
+}
+
+# Immich server
+# Install Immich server and microservices.
+# It asks for informations, builds and installs project, 
+# create a service file and add the machine in hosts list.
+#
+# params: nothing
+#
+# return:
+# - exit if cannot cd in the project directory
+# - return if cannot cd ~
+setup_server()
+{
+    display_message_box "Server & microservices"
+
+    echo "Creating installation directory..."
+    mkdir -p /usr/src/server
+    cd /usr/src/server || exit
+
+    # Build stage
+    echo "Installing build stage dependencies..."
+    apk add --no-cache build-base python3 libheif vips-dev ffmpeg nodejs-current npm
```

Comment:
is `nodejs-current` always the `lts` version of node?

Reply:
Maybe it is better to perform node installation with the version specified in the Dockerfile to avoid unexpected behavior of the application

Reply:
[pkgs.org](https://pkgs.org/search/?q=nodejs) gives packages versions for nodejs on different Alpine versions.
The main `nodejs` package is running node v14, where `nodejs-current` is running v16, as in Dockerfiles.
Alpine 3.14 is going [EOL](https://endoflife.date/alpine) in a few months, I don't think they will update `nodejs-current` over v16.
When Dockerfiles's Alpine version will be updated, we will have to look at the right package with a compatible `node` with Immich.